modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java

```diff
@@ -101,78 +105,137 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
 
 		// See LPS-58543
 
-		String title = "To Do";
+		_addCategory("To Do");
 
-		AssetCategory assetCategory = addCategory(title);
+		_addUser();
 
-		long categoryId = assetCategory.getCategoryId();
+		SearchContext searchContext = _getSearchContext();
 
-		addUser(_group, categoryId);
+		Facet facet = _categoryFacetFactory.newInstance(searchContext);
 
-		SearchContext searchContext = getSearchContext(
-			assetCategory.getTitleCurrentValue());
+		searchContext.addFacet(facet);
 
-		searchContext.setCategoryIds(new long[] {categoryId});
-		searchContext.setGroupIds(new long[] {_group.getGroupId()});
+		FacetsAssert.assertFrequencies(
+			facet.getFieldName(), searchContext, search(searchContext),
+			Collections.singletonMap(_getAssetVocabularyCategoryId(), 1));
+	}
 
-		Facet facet = categoryFacetFactory.newInstance(searchContext);
+	@Test
+	public void testSelection() throws Exception {
+		_addCategory(RandomTestUtil.randomString());
+
+		_addJournalArticle();
+
+		_addUser();
+
+		SearchContext searchContext = _getSearchContext();
+
+		Facet facet = _categoryFacetFactory.newInstance(searchContext);
+
+		facet.select(_getAssetVocabularyCategoryId());
 
 		searchContext.addFacet(facet);
 
 		Hits hits = search(searchContext);
 
-		Map<String, Integer> frequencies = Collections.singletonMap(
-			_getAssetVocabularyCategoryId(assetCategory), 1);
+		_assertEntryClassNames(
+			Collections.singletonList(User.class.getName()), hits,
+			searchContext);
 
 		FacetsAssert.assertFrequencies(
-			facet.getFieldName(), searchContext, hits, frequencies);
+			facet.getFieldName(), searchContext, hits,
+			Collections.singletonMap(_getAssetVocabularyCategoryId(), 1));
 	}
 
-	protected AssetCategory addCategory(String title) throws Exception {
-		AssetCategory assetCategory = assetCategoryLocalService.addCategory(
+	private void _addCategory(String title) throws Exception {
+		_assetCategory = _assetCategoryLocalService.addCategory(
 			TestPropsValues.getUserId(), _group.getGroupId(), title,
 			_assetVocabulary.getVocabularyId(),
 			ServiceContextTestUtil.getServiceContext(
 				_group.getGroupId(), TestPropsValues.getUserId()));
+	}
 
-		_assetCategory = assetCategory;
-
-		return assetCategory;
+	private void _addJournalArticle() throws Exception {
+		journalArticleSearchFixture.addArticle(
+			new JournalArticleBlueprint() {
```

Comment:
Use [JournalArticleBlueprintBuilder](https://github.com/liferay/liferay-portal/blob/master/modules/apps/journal/journal-test-util/src/main/java/com/liferay/journal/test/util/search/JournalArticleBlueprintBuilder.java). See https://github.com/AlmirFe/liferay-portal/commit/2cbd69c19ac68c9912d5fa11408c7ad3a6b4bac1

Reply:
Hi @AlmirFe, for this change I think it's more a matter of preference of each pattern, I'll keep it the way it is to follow the standard, I see that the facets that use this method, use this way, but we can create a ticket to discuss which would be better and standardize.